compiler/rustc_abi/src/lib.rs

```diff
@@ -337,23 +337,21 @@ impl TargetDataLayout {
         Ok(dl)
     }
 
-    /// Returns exclusive upper bound on object size.
+    /// Returns **exclusive** upper bound on object size.
     ///
     /// The theoretical maximum object size is defined as the maximum positive `isize` value.
```

Comment:
I am not entirely sure what is even meant by "object". In general we are saying that "allocated objects" are limited by `isize::MAX`, and we do not publicly document anything else even for 64bit targets. So we do currently allow people to create heap allocations larger than obj_size_bound, and we rely on LLVM supporting that. It is only Rust *types* that can't be any bigger, and therefore `static` and `let`-bound variables and `Box`.

So either we should clarify the comment to say that this does not apply to all heap allocations, or we need to fix our docs.

@nikic will LLVM miscompile things if a heap allocation gets bigger than 2^61 bytes because that overflows `u64` when counting in bits? Or does the 2^64 bits limit only apply to LLVM "typed objects"? Or is this too theoretical because nothing can be that big anyway? ;)

Reply:
I think it would only be a problem if it's a typed allocation/access, but I'm not particularly confident in that :)

Reply:
anyone got a 64-bit address space computer lying around to find out?

Reply:
> I am not entirely sure what is even meant by "object".

Honestly I'm also not completely confident... I think allocation is what is meant?

I was pretty much just rolling with it because "we have a slightly wibbly definition of 'object' deep in the codegen backend" feels kinda low on our list of concerns. Relatively speaking.